packages/devextreme/testing/testcafe/tests/dataGrid/filterPanel.ts

```diff
@@ -0,0 +1,41 @@
+import { Selector } from 'testcafe';
+import url from '../../helpers/getPageUrl';
+import createWidget from '../../helpers/createWidget';
+import DataGrid from '../../model/dataGrid';
+
+fixture.disablePageReloads`filterPanel`
+  .page(url(__dirname, '../container.html'));
+
+const POPUP_CLASS = '.dx-popup';
+const INVISIBLE_CLASS = '.dx-state-invisible';
+
+function getVisiblePopups(): Selector {
+  return Selector(`${POPUP_CLASS}:not(${INVISIBLE_CLASS})`);
+}
```

Comment:
Question.

Maybe we should add this code to the existing popup model as a static method? 
It's live here: packages/devextreme/testing/testcafe/model/popup.ts

In this case other developers can use it in their test cases :)

Reply:
I think we shouldn't 
I wrote this code because writing good test models would require a lot of time. Ideally there should be 

```js
t.expect(filterBuilder.getValueText(0).getEditor().getPopup().element.exists).ok()
```

instead of checking count of all popups. I don't want to persuade other developers to write bad code :)